libs/pubsub/query/query_test.go

```diff
@@ -11,6 +12,47 @@ import (
 	"github.com/cometbft/cometbft/libs/pubsub/query"
 )
 
+func TestBigNumbers(t *testing.T) {
+	testCases := []struct {
+		s        string
+		events   map[string][]string
+		err      bool
+		matches  bool
+		matchErr bool
+	}{
+
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000"}}, false, true, false},
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.0"}}, false, true, false},
+		{"account.balance <= 20000000000000000000", map[string][]string{"account.balance": {"10000000000000000000"}}, false, true, false},
+		{"account.balance <= 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000001"}}, false, false, false},
+		{"account.balance <= 10000000000000000002", map[string][]string{"account.balance": {"10000000000000000000.6"}}, false, true, false},
+		{"account.balance > 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.6"}}, false, true, false},
```

Comment:
Could you add the following checks for negative values?

```suggestion
		{"account.balance > 10000000000000000000", map[string][]string{"account.balance": {"10000000000000000000.6"}}, false, true, false},
				{"account.balance <= -10000000000000000000", map[string][]string{"account.balance": {"-10000000000000000000"}}, false, true, false},
		{"account.balance <= -10000000000000000000", map[string][]string{"account.balance": {"-10000000000000000000.0"}}, false, true, false},
		{"account.balance <= -20000000000000000000", map[string][]string{"account.balance": {"-10000000000000000000"}}, false, false, false},
		{"account.balance <= -10000000000000000000", map[string][]string{"account.balance": {"-10000000000000000001"}}, false, true, false},
		{"account.balance <= -10000000000000000002", map[string][]string{"account.balance": {"-10000000000000000000.6"}}, false, false, false},
		{"account.balance > -10000000000000000000", map[string][]string{"account.balance": {"-10000000000000000000.6"}}, false, false, false},

```

Reply:
Looking at the error I get when I run this and at the rules of the [query grammar](https://github.com/cometbft/cometbft/blob/jasmina/726-pubsub-int-overflow/libs/pubsub/query/query.peg), it seems that negative numbers in the query are not allowed.

Reply:
Thanks for checking. Indeed it does not seem like it. There is no limitation on the data itself, right? Is this a bug?

Reply:
You are right, you could technically submit anything as a tx event I believe. But querying is not supported. I am not sure tbh...in does seem wrong to disable querying something that can be provided as input :/